### PR TITLE
feature: send target app's protocol (http or tcp) when starting wsproxy

### PR DESCRIPTION
**Pull request by kyujin-cho**
This PR adds new protocol variable which defines app's underlying transportation protocol. Sending protocol to wsproxy will help wsproxy determine equivalent proxying mechanism.

### CODE REVIEW

File: src/components/backend-ai-app-launcher.ts [hunk 1]
```diff
@@ -521,14 +521,23 @@ export default class BackendAiAppLauncher extends BackendAIPage {
...
-    const kInfo = await globalThis.backendaiclient.computeSession.get(['scaling_group'], sessionUuid);
+    const kInfo = await globalThis.backendaiclient.computeSession.get(['scaling_group', 'service_ports'], sessionUuid);
     if (kInfo === undefined) {
       this.indicator.end();
       this.notification.text = _text('session.CreationFailed'); // TODO: Change text
 
       this.notification.show();
       return Promise.resolve(false);
     }
+    console.log(kInfo.compute_session.service_ports)
```
Comment (inureyes) [inline src/components/backend-ai-app-launcher.ts]: Please remove the debug lines. :D

**Review by inureyes — COMMENTED** [review]
Do we need to update proxy in the web UI (`src/lib/wsproxy`) ? I think it does not need to accept the protocol since it just uses localhost HTTP, however for the equivalent API interface between web version and local one?

**Comment by kyujin-cho** [comment]
> Do we need to update proxy in the web UI (`src/lib/wsproxy`) ? I think it does not need to accept the protocol since it just uses localhost HTTP, however for the equivalent API interface between web version and local one?

as you mentioned, we can just skip handling `protocol` query parameter on the webUI's wsproxy since we can't handle TCP based proxying on web version. But perhaps we can normalize `manager.js`' TCP app blocking code block with this `protocol` parameter. I'll add a new commit about this feature.

**Comment by inureyes** [comment]
@kyujin-cho  It does not work since application call returns 404.

image

**Review by inureyes — CHANGES_REQUESTED** [review]
It is not working as intended. Needs some polishings.

**Review by inureyes — APPROVED** [review]
LGTM

**Pull request merged**